extensions/KHR/SPV_KHR_bfloat16.asciidoc

```diff
@@ -0,0 +1,206 @@
+It is implementation-defined whether intermediate values are rounded.
+| 16 | *BFloat16TypeKHR*
+|====
+--
+
+Type-Declaration Instructions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Add the following requirement to *OpTypeCooperativeMatrixKHR*:
+
+    If Component Type has a *BFloat16KHR* encoding then *BFloat16CooperativeMatrixKHR* must be declared.
+
+Conversion Instructions
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Add the following paragraph to *OpFConvert*, *OpConvertFToU*, *OpConvertFToS*, *OpConvertSToF*, and *OpConvertUToF*:
+
+    Conversions to or from floating-point values with the `BFloat16KHR` encoding first convert the source value to IEEE754 binary32, and then from IEEE754 binary32 to the target format.
+    When converting from `BFloat16KHR` to IEEE754 binary32, the additional significand bits are padded with 0s, producing the exact same value.
+    When converting from IEEE754 binary32 to `BFloat16KHR`, the rounding mode is defined by the client API.
+
+
+Arithmetic Instructions
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Add the following requirement to *OpDot*:
+
+    If the Result Type (and the component types of Vector 1 and Vector 2) has a *BFloat16KHR* encoding then *BFloat16DotProductKHR* must be declared.
+
+
+Capabilities
+~~~~~~~~~~~~
+
+Modify Section 3.31, Capability, adding this row to the Capability table:
+
+--
+[options="header",cols="1,5,2"]
+|====
+2+^| Capability ^| Implicitly Declares
+| 5116 | *BFloat16TypeKHR* +
+Uses *OpTypeFloat* to specify types with the *BFloat16KHR* <<Floating_Point_Encoding, floating point encoding>> and the use of conversion functions for those types.
+|
+| 5117 | *BFloat16DotProductKHR* +
+Uses vectors with a *Component Type* of *OpTypeFloat* with the *BFloat16KHR* encoding with *OpDot*.
+| *BFloat16TypeKHR*
+| 5118 | *BFloat16CooperativeMatrixKHR* +
+Uses cooperative matrices with a *Component Type* of *OpTypeFloat* with the *BFloat16KHR* encoding.
+| *BFloat16TypeKHR*, *CooperativeMatrixKHR*
+|====
+--
+
+
+Validation Rules
+~~~~~~~~~~~~~~~~
+
+Add the following bullets to section 2.16.1, Universal Validation Rules:
+
+  * Objects with a type that is or includes a floating-point type with the *BFloat16KHR* encoding must only be used with the following instructions:
+  ** https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#_constant_creation_instructions[Constant Creation Instructions]
+  ** https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#_memory_instructions[Memory Instructions]
+  ** https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#_function_instructions[Function Instructions]
+  ** https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#_conversion_instructions[Conversion Instructions]
+  ** https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#_composite_instructions[Composite Instructions]
+  ** OpDecorate
+  ** OpName
```

Comment:
I would have expected Annotation and Debug categories. For example, OpMemberDecorate or OpDecorateId are not covered now. Similarly OpMemberName.

I'd also expect OpExtInst would be needed for proper debug support so I'd think we're missing more categories.

Reply:
I've added the miscellaneous/extension categories too